### DISCUSSION TITLE
Invariant Violation: Encountered an error trying to infer a GraphQL type for: `component: There is no corresponding node with the `id` field matching

### DISCUSSION THREAD

**Discussion by spond-canoo** [gatsbyjs/gatsby#35031] [Help]
Getting this error when trying to POST to __refresh endpoint on Gatsby-source-contentful development server.

Missing onError handler for invocation 'building-schema', error was 'Invariant Violation: Encountered an error trying to infer a GraphQL type for: `component:
video___NODE`. There is no corresponding node with the `id` field matching: "7c7b7o0fmMBybG2ZP6rXCH".'. Stacktrace was 'Invariant Violation: Encountered an error trying to
infer a GraphQL type for: `component: video___NODE`. There is no corresponding node with the `id` field matching: "7c7b7o0fmMBybG2ZP6rXCH".

**Comment by codebymark** [comment]
@spond-canoo 
I too experienced this issue just now. I was able to resolve it by rebuilding my Gatsby site but first running:
```
gatsby clean
```

I hope this helps

**Reply by spond-canoo** [reply]
Thanks for your answer, although I am a little confused how this would work in the context of running a gatsby preview server.

I have found that I can try / catch the error in the buildSchema, eg:



```
// build-schema.fixed.js
await buildSchema(...arguments);
  } catch (e) {
    console.warn(e);

    // TODO: need special handling.  I don't think I can do a gatsby clean here??

    // await buildSchema(...arguments);
  }
};

```

**Comment by spond-canoo** [comment]
I found a variable that seems to help.  This prevents the rebuilds during a gatsby develop session.  My guess is that the "Experimental_schem_rebuild" Is buggy, but Gatsby enables it by default.

GATSBY_EXPERIMENTAL_DISABLE_SCHEMA_REBUILD=true

**Reply by calebc01** [reply]
> I found a variable that seems to help. This prevents the rebuilds during a gatsby develop session. My guess is that the "Experimental_schem_rebuild" Is buggy, but Gatsby enables it by default.
> 
> GATSBY_EXPERIMENTAL_DISABLE_SCHEMA_REBUILD=true

This worked for me.  I don't recall what caused the original problem for me, but the result was that it was impossible to rebuild content without running "gatsby clean" and then "gatsby develop" again.  Setting this environment variable prevented the issue from recurring.

**Comment by keroles816** [comment]
you found the solution cause I'm didn't have the sol
